core/src/main/scala/org/apache/spark/deploy/master/Master.scala

```diff
@@ -725,26 +729,38 @@ private[deploy] class Master(
    */
   private def startExecutorsOnWorkers(): Unit = {
     // Right now this is a very simple FIFO scheduler. We keep trying to fit in the first app
-    // in the queue, then the second app, etc.
+    // in the queue, then the second app, etc. And for each app, we will schedule base on
+    // resource profiles also with a simple FIFO scheduler, resource profile with smaller id
+    // first.
```

Comment:
Yes, you are right. Smaller id does not indicate higher priority directly. The strategy is simple and straightforward, relying on the dynamic allocation manager to adjust the target number for each resource profile, and we just fulfill the requirements in a fixed order.

Currently, we don't have a good way to infer about the order of requests for different resource profiles. The information we have are the target number and existing executor number for each resource profile. Maybe we can compute the priority somehow in a FAIR schedule mode.

Reply:
> Currently, we don't have a good way to infer about the order of requests for different resource profiles.

I actually means the order of receiving the request in Master, although I know it could be out of order compared to the request sender (driver) due to asynchronous RPC framework. But after a second thinking, requests come from the pending tasks, which are able to be scheduled in parallel as long as there're enough resources. So it doesn't really matter which resource profile should be used to launch executors. Schedule by ordered resource profile ids should be enough.